lib/server.js

```diff
@@ -102,17 +101,21 @@ function createServer (options, httpHandler) {
           return new Promise((resolve, reject) => {
             multipleBindings.call(this, server, httpHandler, options, listenOptions, () => {
               this[kState].listening = true
+              this[kAvvioBoot](() => {})
```

Comment:
I believe that here you are covering pretty well the scenarios when `localhost` is used, but I'm pretty sure we might be missing when other than `localhost` is used (e.g. `127.0.0.1` or `::1`)

Reply:
This is a very good point, which we have been working on and will focus on this now.

Reply:
How does this look? It seems to function properly (passing all tests) but I feel iffy about it because now for non-localhosts I use `then` which may not function as desired.
```
    if (cb === undefined) {
      const listening = listenPromise.call(this, server, listenOptions)
      /* istanbul ignore else */

      return listening.then(address => {
        return new Promise((resolve, reject) => {
          function handleOnListenError (err) {
            if (err) {
              return reject(err)
            }
          }
          if (host === 'localhost') {
            multipleBindings.call(this, server, httpHandler, options, listenOptions, () => {
              this[kState].listening = true
              this[kAvvioBoot](() => {})
              hookRunnerApplication('onListen', this[kAvvioBoot], this, handleOnListenError)
              resolve(address)
            })
          } else {
            hookRunnerApplication('onListen', this[kAvvioBoot], this, handleOnListenError)
            resolve(address)
          }
        })
      })
    }
    this.ready(listenCallback.call(this, server, listenOptions))
```

Reply:
The `listenPromise` is built by us, so it should be deterministic. If it fails, we follow the rule of checking either host or anything else as the initial listening failed.

The code LGTM 👍